huaweicloud/services/iotda/resource_huaweicloud_iotda_access_credential.go

```diff
@@ -0,0 +1,108 @@
+package iotda
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/iotda/v5/model"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API IoTDA POST /v5/iot/{project_id}/auth/accesscode
+func ResourceAccessCredential() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAccessCredentialCreate,
+		ReadContext:   resourceAccessCredentialRead,
+		DeleteContext: resourceAccessCredentialDelete,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"force_disconnect": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				ForceNew: true,
+			},
+			"access_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"access_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildAccessCredentialBodyParams(d *schema.ResourceData) *model.CreateAccessCodeRequest {
+	bodyParams := model.CreateAccessCodeRequest{
+		Body: &model.CreateAccessCodeRequestBody{
+			Type:            utils.StringIgnoreEmpty(d.Get("type").(string)),
+			ForceDisconnect: utils.Bool(d.Get("force_disconnect").(bool)),
+		},
+	}
+
+	return &bodyParams
+}
+
+func resourceAccessCredentialCreate(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	isDerived := WithDerivedAuth(cfg, region)
+	client, err := cfg.HcIoTdaV5Client(region, isDerived)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	createOpts := buildAccessCredentialBodyParams(d)
+	respBody, err := client.CreateAccessCode(createOpts)
+	if err != nil || respBody == nil {
+		return diag.Errorf("error creating access credential: %s", err)
+	}
+
+	resourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+
+	d.SetId(resourceId)
+
+	accessKey := utils.StringValue(respBody.AccessCode)
+	accessCode := utils.StringValue(respBody.AccessKey)
+	if accessKey == "" || accessCode == "" {
+		return diag.Errorf("error creating access credential: 'access_key' or 'access_code' in API response: %s", err)
```

Comment:
It makes no sense to print err here.
```suggestion
		return diag.Errorf("error creating access credential: 'access_key' or 'access_code' is empty in API response")
```